src/main.ts

```diff
@@ -4,15 +4,35 @@ import OpenAI from "openai";
 import { Octokit } from "@octokit/rest";
 import parseDiff, { Chunk, File } from "parse-diff";
 import minimatch from "minimatch";
+import { Certificate } from "crypto";
 
 const GITHUB_TOKEN: string = core.getInput("GITHUB_TOKEN");
 const OPENAI_API_KEY: string = core.getInput("OPENAI_API_KEY");
 const OPENAI_API_MODEL: string = core.getInput("OPENAI_API_MODEL");
+const OPENAI_API_VERSION: string = core.getInput("OPENAI_API_VERSION");
+const OPENAI_BASE_URL: string = core.getInput("OPENAI_BASE_URL"); // Keep the default value as undefined instead of empty strings.
+
+// Supports HTTP requests debugging via an environment variable.
+const debugHttp: string | undefined = process.env.DEBUG_HTTP;
+if (debugHttp) {
+  // Intercept all HTTP requests
+  const nock = require("nock");
+  nock.recorder.rec({
+    output_objects: true,
+    logging: (content: any) => {
+      console.log("HTTP Request:", content);
+    },
+  });
+  console.log("HTTP calls interception enabled");
+}
 
 const octokit = new Octokit({ auth: GITHUB_TOKEN });
 
 const openai = new OpenAI({
   apiKey: OPENAI_API_KEY,
+  baseURL: OPENAI_BASE_URL,
+  defaultQuery: { "api-version": OPENAI_API_VERSION },
+  defaultHeaders: { "api-key": OPENAI_API_KEY },
```

Comment:
The `defaultQuery` and `defaultHeaders` are required when interacting with Azure OpenAI API and a deviation with OpenAI API which does not require these.